mis_builder_budget/models/mis_report_instance_period.py

```diff
@@ -32,7 +32,25 @@ def _get_additional_budget_item_filter(self):
         to be inherited, and is useful to implement filtering
         on analytic dimensions or operational units.
 
+        The default filter is built from a ``mis_report_filters context``
+        key, which is a list set by the analytic filtering mechanism
+        of the mis report widget::
+
+          [(field_name, {'value': value, 'operator': operator})]
+
+        This default filter is the same as the one set by
+        _get_additional_move_line_filter on mis.report.instance, so
+        a budget.item is expected to have the same analytic fields as
+        a move line.
+
         Returns an Odoo domain expression (a python list)
         compatible with mis.budget.item."""
         self.ensure_one()
-        return []
+        filters = []
+        mis_report_filters = self.env.context.get('mis_report_filters', {})
+        for filter_name, values in mis_report_filters.items():
+            if values:
+                value = values.get('value')
+                operator = values.get('operator', '=')
+                filters.append((filter_name, operator, value))
+        return filters
```

Comment:
I think we should have a single method which get the analytic filters from the context, here we have the logic in ´``mis_builder/models/mis_report_instance``` and ´``mis_builder_budget/models/mis_report_instance_period```. What do you think @sbidoul?

Reply:
@benwillig indeed, it would be better. Can you do it?

Reply:
Done